vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridSingleSelectionIT.java

```diff
@@ -85,4 +85,25 @@ public void checkDeselectionDisallowedInitially() {
         Assert.assertTrue("Row 1 was still selected after de-selecting it.",
                 !grid.getRow(1).isSelected());
     }
+
+    @Test
+    public void gridSetItemsAndSelectItemAndSetItemAfterSetDeselectAllowedFalse() {
+        open();
+
+        // De-selection is allowed(deselectAllowed is false) and then setting items for grid
+        GridElement grid = $(GridElement.class)
+                .id(GridSingleSelectionPage.ITEMS_GRID);
+
+        // Set Items again by clicking the button
+        $("button").id(GridSingleSelectionPage.SET_ITEMS)
```

Comment:
The button seems to do a lot more than just setting the items = it also calls `removeAllColumns`, `addColumn`, `setHeightByRows`, `setId` and also re-adds the whole grid (so it ends up being located after the button)

Reply:
separated the method